dask/dataframe/io/parquet.py

```diff
@@ -174,11 +147,147 @@ def _read_parquet_row_group(open, fn, index, columns, rg, series, categories,
         return df
 
 
+# ----------------------------------------------------------------------
+# PyArrow interface
+
+def _read_pyarrow(fs, paths, file_opener, columns=None, filters=None,
+                  categories=None, index=None):
+    if not pyarrow_parquet:
+        raise ImportError("pyarrow.parquet not installed")
+
+    api = pyarrow_parquet
+
+    if filters is not None:
+        raise NotImplemented("Predicate pushdown not implemented")
+
+    if categories is not None:
+        raise NotImplemented("Categorical reads not yet implemented")
+
+    if isinstance(columns, tuple):
+        columns = list(columns)
+
+    columns = columns
+
+    dataset = api.ParquetDataset(paths)
+    schema = dataset.schema.to_arrow_schema()
+    task_name = 'read-parquet-' + tokenize(dataset, columns)
+
+    pieces = dataset.pieces
+
+    if columns is None:
+        all_columns = schema.names
+    else:
+        all_columns = columns
+
+    if not isinstance(all_columns, list):
+        out_type = Series
+        all_columns = [all_columns]
+    else:
+        out_type = DataFrame
+
+    divisions = (None,) * (len(dataset.pieces) + 1)
+
+    dtypes = _get_pyarrow_dtypes(schema)
+
+    meta = _meta_from_dtypes(all_columns, schema.names, dtypes)
+
+    task_plan = {
+        (task_name, i): (_read_arrow_parquet_piece,
+                         file_opener,
+                         piece, all_columns,
+                         out_type == Series,
+                         dataset.partitions)
+        for i, piece in enumerate(pieces)
+    }
+
+    return out_type(task_plan, task_name, meta, divisions)
+
+
+def _get_pyarrow_dtypes(schema):
+    dtypes = {}
+    for i in range(len(schema)):
+        field = schema[i]
+        numpy_dtype = field.type.to_pandas_dtype()
+        dtypes[field.name] = numpy_dtype
+
+    return dtypes
+
+
+def _read_arrow_parquet_piece(open_file_func, piece, columns, is_series,
+                              partitions):
+    with open_file_func(piece.path, mode='rb') as f:
+        table = piece.read(columns=columns,  partitions=partitions,
+                           file=f)
+    df = table.to_pandas()
+
+    if is_series:
+        return df[df.columns[0]]
+    else:
+        return df
+
+
+# ----------------------------------------------------------------------
+# User read API
+
+def read_parquet(path, columns=None, filters=None, categories=None, index=None,
+                 storage_options=None, engine='fastparquet'):
```

Comment:
We could change `engine=` to `'auto'` that would use whichever library was present.